android/library/src/main/java/com/polidea/multiplatformbleadapter/BleModule.java

```diff
@@ -255,8 +258,11 @@ public void readRSSIForDevice(String deviceIdentifier,
                                   final String transactionId,
                                   final OnSuccessCallback<Device> onSuccessCallback,
                                   final OnErrorCallback onErrorCallback) {
-        final Device device = getDeviceOrEmitError(deviceIdentifier, onErrorCallback);
-        if (device == null) {
+        final Device device;
+        try {
+            device = getDeviceById(deviceIdentifier);
+        } catch (BleError error) {
+            onErrorCallback.onError(error);
             return;
         }
         final RxBleConnection connection = getConnectionOrEmitError(device.getId(), onErrorCallback);
```

Comment:
Should we also simplify `getConnectionOrEmitError`?

Reply:
I added issue for that.